### ISSUE TITLE
Add individual delay for each file 

### ISSUE THREAD

**Issue by Toxenskiy** [UnamSanctam/UnamBinder#7]

**Comment by Toxenskiy** [comment]
?

**Comment by UnamSanctam** [comment]
Well, the problem is that it will of course delay it all sequentially (since that's how programming works, and of course I'm not about to make it multithreaded just for that), so that let's say you have 3 files and a 20 seconds delay selected on each file, then the delay on the first will be 20 total seconds, the delay on the second file would be 40 total seconds and the delay for the 3rd would be 60 total seconds. So the third and final file would be extracted and started 60 seconds after the binder file was started. Due to this I would probably also have to create up and down buttons that moves files in the list since it would execute in that order, it's also a bit difficult on how I will present it in a way that it's more obvious that it will execute sequentially for those who don't know anything about programs.

**Comment by Toxenskiy** [comment]
Is there a problem with this implementation?

**Comment by UnamSanctam** [comment]
It will be difficult to convey in a simple manner to the user which is why I have delayed the implementation of it until I can figure out a good way to convey it. Most would probably not also understand that if the file is killed (by something like the Task Manager) while it is waiting then it won't drop and execute and further files (since it's not running) which I would also have to convey, maybe I can do it in the wiki but most people don't read it.

**Comment by Toxenskiy** [comment]
Solved with a simple switch ..

**Comment by Toxenskiy** [comment]
Or the "advanced" tab.

**Comment by UnamSanctam** [comment]
Yes but how to convey it is the problem, if a normal user sets a 20 second delay on all files they would expect all the files to drop and execute after 20 seconds, but that's not how it would work in practice of course but would instead drop and execute them in intervals of 20 seconds over 60 seconds.

**Comment by Toxenskiy** [comment]
we divide the delay by the number of files and run each file through the result

**Comment by Toxenskiy** [comment]
and for interested users, do a deeper customization.

**Comment by UnamSanctam** [comment]
Well, with that implementation that delay would make even less sense for the user, I'll have to figure out some good way to indicate how it works to the user or I might just have to implement it in an ugly way (the delays are sequential while not obviously indicated as such) since I do not really want to create or implement asynchronous functions or multithreading though I'd much rather come up with some good solution.

**Comment by Toxenskiy** [comment]
if I'm not mistaken c # is the simplest language for asynchrony.

**Comment by UnamSanctam** [comment]
Yes of course but the binder and downloader is in C since I want to generate native files because C# files (the code) will eventually always be detected since they are managed assemblies and it will require some specific version of .NET. Creating "malware" in C# is generally a bad idea compared to languages like C or C++ though it's a lot lot easier in C# which is why it's usually the language used.

**Comment by Toxenskiy** [comment]
And I forgot that the build itself is in c ++ / c, well, you still have to make antivm.

**Comment by Toxenskiy** [comment]
you can give me the delay code before starting the miner

**Comment by Toxenskiy** [comment]
?

**Comment by UnamSanctam** [comment]
What do you mean?

**Comment by Toxenskiy** [comment]
Well, I'm talking about the miner, I'm just talking in the wrong repository.

**Comment by Toxenskiy** [comment]
found

![image](https://user-images.githubusercontent.com/55586522/134541352-64b09688-6773-4ce9-829c-e44b4b3b6ad8.png)

**Comment by Toxenskiy** [comment]
changed the code a little, the detections are lower;)

**Comment by UnamSanctam** [comment]
Yes, making it more unique will do that since it will not be as close to the source code.

**Comment by Toxenskiy** [comment]
+

**Comment by Toxenskiy** [comment]
By the way, how are you doing with the miner?

**Comment by Toxenskiy** [comment]
I have a question, after the compilation there are some traces by which you can find you on github or some other sources?

**Comment by UnamSanctam** [comment]
Not really except maybe "Online Download" since it contains the links to the miner (though it's encrypted).

**Comment by Toxenskiy** [comment]
Well, you yourself know how requests are tracked.
